Flagsmith.FlagsmithClient/FlagsmithConfiguration.cs

```diff
@@ -1,37 +1,72 @@
 ﻿using Microsoft.Extensions.Logging;
 using System;
 using System.Collections.Generic;
+using OfflineHandler;
+using System.Net.Http;
 
 namespace Flagsmith
 {
-    public class FlagsmithConfiguration : IFlagsmithConfiguration
+    public class FlagsmithConfiguration
     {
-        public FlagsmithConfiguration()
+        private static readonly Uri DefaultApiUri = new Uri("https://edge.api.flagsmith.com/api/v1/");
+        private TimeSpan _timeout;
+
+        /// <summary>
+        /// <para>Override the URL of the Flagsmith API to communicate with.</para>
+        /// <para>Deprecated since 7.1.0. Use <see cref="ApiUri"/> instead.</para>
+        /// </summary>
+        [Obsolete("Use ApiUri instead.")]
+        public string ApiUrl
         {
-            ApiUrl = "https://edge.api.flagsmith.com/api/v1/";
-            EnvironmentKey = string.Empty;
-            EnvironmentRefreshIntervalSeconds = 60;
+            get => ApiUri.ToString();
+            set => ApiUri = value.EndsWith("/") ? new Uri(value) : new Uri($"{value}/");
         }
+
         /// <summary>
-        /// Override the URL of the Flagsmith API to communicate with.
+        /// Versioned base Flagsmith API URI to use for all requests. Defaults to
+        /// <c>https://edge.api.flagsmith.com/api/v1/</c>.
+        /// <example><code>new Uri("https://flagsmith.example.com/api/v1/")</code></example>
         /// </summary>
-        public string ApiUrl { get; set; }
+        public Uri ApiUri { get; set; } = DefaultApiUri;
```

Comment:
Since the initial point of this PR was to bring consistency between the server side SDKs (at least that's my understanding), I'm surprised to see us change the name of this argument? Why not just keep using the string and build the URI ourselves?

Reply:
Making server-side SDKs consistent in their parameter names was the initial driver for this PR. But, because the SDK constructor uses position-based and named arguments we can't remove or rename any arguments without introducing breaking changes.

By accepting the URI as an object and not a string, we don't have to deal with parsing URLs manually or error handling. FxCop has this as a default rule to enforce exactly this: https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1054.

My opinion is, if there's already a universally-used type for dealing with URIs, and it takes almost no effort to use, there's no reason to have to manipulate strings ourselves. And of course, this is exactly how our favourite competitor implemented this in their SDK.

Reply:
This is also the same reason why I deprecated `int EnvironmentRefreshIntervalSeconds` in favour of `TimeSpan EnvironmentRefreshInterval`.

Reply:
Yep, ok, fair enough.